cloudferrylib/os/compute/nova_compute.py

```diff
@@ -583,6 +583,27 @@ def attach_volume_to_instance(self, instance, volume):
     def dissociate_floatingip(self, instance_id, floatingip):
         self.nova_client.servers.remove_floating_ip(instance_id, floatingip)
 
+    def get_hypervisor_statistics(self):
+        return self.nova_client.hypervisors.statistics()
+
     def get_hypervisors(self):
         return [hypervisor.host_ip for hypervisor in
                 self.nova_client.hypervisors.list()]
+
+    def get_free_vcpus(self):
+        hypervisor_statistics = self.get_hypervisor_statistics()
+        return hypervisor_statistics.vcpus * float(
+            self.config.compute.cpu_allocation_ratio) - \
+               hypervisor_statistics.vcpus_used
+
+    def get_free_ram(self):
+        hypervisor_statistics = self.get_hypervisor_statistics()
+        return hypervisor_statistics.memory_mb * float(
+            self.config.compute.ram_allocation_ratio) - \
+               hypervisor_statistics.memory_mb_used
+
+    def get_free_disk(self):
+        hypervisor_statistics = self.get_hypervisor_statistics()
+        return hypervisor_statistics.local_gb * float(
+            self.config.compute.disk_allocation_ratio) - \
```

Comment:
Could we do not use such line breaks please?

According to the pep8 (https://www.python.org/dev/peps/pep-0008/#maximum-line-length): 
The preferred way of wrapping long lines is by using Python's implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.